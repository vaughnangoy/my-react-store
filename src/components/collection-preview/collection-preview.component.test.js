import React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import CollectionPreview from './collection-preview.component';

jest.mock('../collection-item/collection-item.component', () => () => (
    <div data-testid="mock-collection-item" />
));

describe('CollectionPreview', () => {
    afterEach(cleanup);

    it('renders', () => {
        const { asFragment } = render(<CollectionPreview />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should display 3 children', () => {
        const props = {
            title: 'Test Item Title',
            items: [{ id: 1 }, { id: 2 }, { id: 3 }],
        };
        const { getAllByTestId } = render(<CollectionPreview {...props} />);
        expect(getAllByTestId('mock-collection-item').length).toEqual(3);
    });
});
