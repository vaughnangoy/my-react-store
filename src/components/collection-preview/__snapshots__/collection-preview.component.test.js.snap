// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview renders 1`] = `
<DocumentFragment>
  <div
    class="collection-preview"
  >
    <h1
      class="title"
    />
    <div
      class="preview"
    />
  </div>
</DocumentFragment>
`;
